FF1Lib/Weapon.cs

```diff
@@ -200,6 +200,420 @@ public void MagisizeWeapons(MT19337 rng, bool balanced)
+
+			    int weaponType;
+			    switch (weaponItemId) {
+				case Item.Masamune:
+				case Item.Xcalber:
+				    // 0 (sword) or 1 (axe)
+				    weaponType = rng.Between(0, 1);
+				    break;
+				case Item.Katana:
+				case Item.Vorpal:
+				case Item.Defense:
+				    weaponType = 0;
+				    break;
+				default:
+				    if (requireTypes.Count > 0) {
+					weaponType = requireTypes[0];
+				    } else {
+					weaponType = rng.Between(0, 5);
+				    }
+				    break;
+			    }
+
+			    icon = weaponIcons[weaponType];
+
+			    int damageTier = tier + weaponTypeAdjust[weaponType, 0];
+			    int critTier =   tier + weaponTypeAdjust[weaponType, 1];
+			    int hitpctTier = tier + weaponTypeAdjust[weaponType, 2];
+
+			    if (weaponItemId == Item.Katana || weaponItemId == Item.Vorpal) {
+				critTier += 1;
+			    }
+
+			    if (weaponItemId == Item.Masamune) {
+				critTier -= 1;
+			    }
+
+			    if (weaponItemId == Item.Vorpal) {
+				damageTier -= 1;
+			    }
+
+			    damageTier = Math.Min(Math.Max(damageTier, 0), 4);
+			    critTier   = Math.Min(Math.Max(critTier,   0), 4);
+			    hitpctTier = Math.Min(Math.Max(hitpctTier, 0), 4);
+
+			    damage = (byte)RangeScale(damageBases[damageTier], .7, 1.4, 1.0, rng);
+			    crit = (byte)RangeScale(critPercentBases[critTier], .7, 1.4, 1.0, rng);
+			    hitBonus = (byte)RangeScale(hitPercentBases[hitpctTier], .7, 1.4, 1.0, rng);
+
+			    // need to clamp this to prevent overflow
+			    hitBonus = Math.Min(hitBonus, (byte)50);
+
+			    double ddamage = damage;
+			    double dcrit = crit;
+			    double dhitBonus = hitBonus;
+
+			    // "score" is a rough approximation of DPS
+			    // used to assign name based on gear
+			    // quality and set the base price.
+			    double score = ((ddamage*1.5) + ((ddamage*2.0) * (dcrit / 200.0))) * (1+Math.Floor((dhitBonus+4)/32));
+
+			    if (weaponItemId == Item.Defense) {
+				var defMagic = defenseSwordSpells[rng.Between(0, defenseSwordSpells.Length-1)];
+				for (int i = 0; i < Spells.Count; i++) {
+				    if (Spells[i].Name == defMagic) {
+					spellIndex = (byte)i;
+					break;
+				    }
+				}
+			    } else {
+				int spellChance = rng.Between(1, 100);
+				if ((weaponType < 4 && spellChance <= 20)
+				    || (weaponType >= 4 && spellChance <= 50))
+				{
+				    var spelllevelLow = Math.Min(tier*2, 6); // 2, 4, 6, 6
+				    var spelllevelHigh = Math.Min(spelllevelLow+4, 8); // 6, 8, 8, 8
+				    do {
+					spellIndex = (byte)rng.Between(spelllevelLow*8, spelllevelHigh*8-1);
+				    } while(Spells[spellIndex].Data[4] == 0); // must be combat castable
+				}
+			    }
+
+			    // Need to special case: Defense, Katana, Xcal, Vorpal, Masa
+
+			    int specialPower = -1;
+			    if (spellIndex == 0xFF) {
+				int powerChance = rng.Between(1, 100);
+				if (tier == 1 && powerChance <= 20) {
+				    specialPower = rng.Between(0, powers.Length-1);
+				} else if (tier > 1 && powerChance <= 80) {

```

Comment:
Isn't that a little high?

Reply:
This means 20% of common weapons will have extra powers, and 80% of rare weapons will.  In vanilla, 100% of rare weapons have extra powers, that's what makes them rare.  Are you suggesting it's too high for common items?

Reply:
80% seemed a bit high, but with your explanation, it's good